.github/workflows/binary-applications.yml

```diff
@@ -45,22 +45,16 @@ jobs:
             - name: Fetch git tags
               run: git fetch origin +refs/tags/*:refs/tags/*
 
-            - name: Restore pip cache
-              uses: actions/cache@v2
+            - name: Set up python and conda environment
+              uses: ./.github/actions/setup_env
               with:
-                  path: ~\AppData\Local\pip\Cache
-                  key: windows-py${{ env.PYTHON_VERSION }}-${{ env.PYTHON_ARCH }}-pip-${{ hashFiles('**/requirements*.txt') }}-exe
-
-            - name: Setup conda environment
-              uses: conda-incubator/setup-miniconda@v2
-              with:
-                  activate-environment: winbin-env
-                  auto-update-conda: true
-                  python-version: 3.8
-                  channels: conda-forge
+                  key: ${{ env.PYTHON_VERSION }}-windows-binaries
+              id: condacache
 
             - name: Install python dependencies
+              if: steps.condacache.outputs.cache-hit != 'true'
               run: |
+                  conda install python=${{ env.PYTHON_VERSION }}

```

Comment:
Install python with other dependencies, rather than in `setup-miniconda` action. That way the python installation gets cached and reused as part of the environment too.